tensorflow_lite_support/ios/task/processor/sources/TFLSegmentationResult.h

```diff
@@ -0,0 +1,129 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ==============================================================================*/
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** Holds a confidence mask belonging to a single class and its meta data. */
+@interface TFLConfidenceMask : NSObject
+
+/**
+ * Confidence masks of size `width` x `height` for any one class.
+ */
+@property(nonatomic, assign) float *mask;
+
+/**
+ * The width of the mask. This is an intrinsic parameter of the model being
+ * used, and does not depend on the input image dimensions.
+ */
+@property(nonatomic, assign) NSInteger width;
+
+/**
+ *  The height of the mask. This is an intrinsic parameter of the model being
+ * used, and does not depend on the input image dimensions.
+ */
+@property(nonatomic, assign) NSInteger height;
+
+@end
+
+/** Holds category mask and its metadata. */
+@interface TFLCategoryMask : NSObject
+
+/**
+ * Flattened 2D-array of size `width` x `height`, in row major order.
+ * The value of each pixel in this mask represents the class to which the
+ * pixel belongs.
+ */
+@property(nonatomic, assign) NSUInteger *mask;
```

Comment:
@priankakariatyml As we want to optimize for performance when dealing with copying large mask, let's use the same datatype as the C (uint8_t) so that we can use memcpy rather than copying and casting each element.

Sorry for the back-and-forth.

Reply:
Fixed